lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -2244,7 +2245,15 @@ void ChargePointImpl::sign_certificate(const ocpp::CertificateSigningUseEnum& ce
         this->configuration->getCpoName().value(), this->configuration->getChargeBoxSerialNumber(),
         this->configuration->getUseTPM());
 
-    req.csr = csr;
+    if (!csr.has_value()) {
+        EVLOG_error << "Create CSR (TPM=" << this->configuration->getUseTPM() << ")"
```

Comment:
Also here Id prefer to send a SecurityEventNotification.req with `CSRGenerationFailed`